CMakeLists.txt

```diff
@@ -1,6 +1,5 @@
 
 # Options
-
 option(SLEEF_BUILD_STATIC_TEST_BINS "Build statically linked test executables" OFF)
```

Comment:
The behavior of `option` and `set` are controlled by policies. You _must_ set `cmake_minimum_required` as the first line of your top-level program to have _any_ backwards compatibility guarantees at all.

Ideally `project()` would be the second line since _so many_ other commands depend on the internal state it configures.

```suggestion
cmake_minimum_required(VERSION 3.18)
project(SLEEF LANGUAGES C VERSION 3.6.0)

# TODO: rename uses of this variable to use the standard one
set(SLEEF_VERSION_PATCHLEVEL ${SLEEF_VERSION_PATCH})
set(SLEEF_SOVERSION ${SLEEF_VERSION_MAJOR})

option(SLEEF_BUILD_STATIC_TEST_BINS "Build statically linked test executables" OFF)
```

Then, after your options, add the following lines:
```
set(
if (SLEEF_ENABLE_CXX)
  enable_language(CXX)
endif ()
if (SLEEF_ENABLE_CUDA)
  enable_language(CUDA)
endif ()
```

No need to maintain that `LANGLIST` anymore. Just enable the requested languages.

Then, below, delete these lines:

```
cmake_minimum_required(VERSION 3.18)
if(${CMAKE_VERSION} VERSION_GREATER "3.14.99")
  cmake_policy(SET CMP0091 NEW)
endif()
```

`CMP0091` is enabled in CMake 3.15+ so the `cmake_policy` line is a noop and so the whole `if ()` block can be removed.

Also delete these lines:

```
set(SLEEF_VERSION_MAJOR 3)
set(SLEEF_VERSION_MINOR 6)
set(SLEEF_VERSION_PATCHLEVEL 0)
set(SLEEF_VERSION ${SLEEF_VERSION_MAJOR}.${SLEEF_VERSION_MINOR}.${SLEEF_VERSION_PATCHLEVEL})
set(SLEEF_SOVERSION ${SLEEF_VERSION_MAJOR})
set(LANGLIST C)
if (SLEEF_ENABLE_CUDA)
  set(LANGLIST ${LANGLIST} CUDA)
endif()
if (SLEEF_ENABLE_CXX)
  set(LANGLIST ${LANGLIST} CXX)
endif()
```